### PR TITLE
entgql: expose the Entity connection on the Query type

### PR DESCRIPTION
**Pull request by giautm**
Related issue: https://github.com/ent/ent/issues/2351

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -147,7 +147,6 @@ github.com/mattn/go-isatty v0.0.8/go.mod h1:Iq45c/XA43vh69/j3iqttzPXn0bhXyGjM0Hd
 github.com/mattn/go-isatty v0.0.12/go.mod h1:cbi8OIDigv2wuxKPP5vlRcQ1OAZbq2CE4Kysco4FUpU=
```
Comment (giautm) [inline go.sum:147]: Someone forgot to run `go mod` after running `go generate`. So the content of this file is not correct.

File: entgql/annotation.go [hunk 2]
```diff
@@ -45,6 +45,8 @@ type (
 		Implements []string `json:"Implements,omitempty"`
 		// Directives to add on the field/type.
 		Directives []Directive `json:"Directives,omitempty"`
+		// QueryField allow you expose the Entity on the query as a field.
```
Comment (a8m) [inline entgql/annotation.go:48]:
```suggestion
		// QueryField exposes the generated type with the given string under the Query object.
```

Can we also make the name optional, so a call to `QueryField()` will expose the type with generated name (plural form)?

**Comment by a8m** [comment]
Looks great @giautm! Before we merge this, please see my comment.

File: entgql/annotation.go [hunk 3]
```diff
@@ -212,6 +240,24 @@ func (f SkipMode) Is(mode SkipMode) bool {
 	return f&mode != 0
 }
 
+func (c QueryFieldConfig) fieldName(gqlType string) string {
+	if c.Name != "" {
+		return c.Name
+	}
+
```
Comment (a8m) [inline entgql/annotation.go:247]:
```suggestion
```

File: entgql/annotation.go [hunk 4]
```diff
@@ -62,6 +64,13 @@ type (
 
 	// SkipMode is a bit flag for the Skip annotation.
 	SkipMode int
+
+	QueryFieldConfig struct {
```
Comment (a8m) [inline entgql/annotation.go:68]:
```suggestion
	FieldConfig struct {
```
Allow using it in other types in the future (e.g. Mutation).

**Pull request merged**